pallets/automation-time/src/types.rs

```diff
@@ -185,3 +202,114 @@ impl<AccountId> TaskHashInput<AccountId> {
 		Self { owner_id, provided_id }
 	}
 }
+
+#[derive(Debug, Decode, Eq, Encode, PartialEq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct ScheduledTasks<AccountId, TaskId> {
+	pub tasks: Vec<(AccountId, TaskId)>,
+	pub weight: u128,
+}
+impl<A, B> Default for ScheduledTasks<A, B> {
+	fn default() -> Self {
+		Self { tasks: vec![], weight: 0 }
+	}
+}
+impl<AccountId, TaskId> ScheduledTasks<AccountId, TaskId> {
+	pub fn try_push<T: Config, Balance: AtLeast32BitUnsigned>(
+		&mut self,
+		task_id: TaskId,
+		task: &Task<AccountId, Balance, T::CurrencyId, T::MaxExecutionTimes>,
+	) -> Result<&mut Self, Error<T>>
+	where
+		AccountId: Clone,
+	{
+		let weight = self
+			.weight
+			.checked_add(task.action.execution_weight::<T>() as u128)
+			.ok_or(Error::<T>::TimeSlotFull)?;
+		if self.tasks.len() as u32 >= T::MaxTasksPerSlot::get() ||
```

Comment:
Added MaxTasksPerSlot to avoid the risk of too many tasks being scheduled before we understand performance implications.

Reply:
probably worth a comment on why it is still needed